src/API/Google/MerchantReport.php

```diff
@@ -44,12 +59,114 @@ class MerchantReport implements OptionsAwareInterface {
 	/**
 	 * Merchant Report constructor.
 	 *
-	 * @param ShoppingContent $service
-	 * @param ProductHelper   $product_helper
+	 * @param ShoppingContent   $service
+	 * @param ProductHelper     $product_helper
+	 * @param ProductRepository $product_repository
 	 */
-	public function __construct( ShoppingContent $service, ProductHelper $product_helper ) {
-		$this->service        = $service;
-		$this->product_helper = $product_helper;
+	public function __construct( ShoppingContent $service, ProductHelper $product_helper, ProductRepository $product_repository ) {
+		$this->service            = $service;
+		$this->product_helper     = $product_helper;
+		$this->product_repository = $product_repository;
+	}
+
+	/**
+	 * Get product statuses.
+	 *
+	 * @return array List of product statuses.
+	 * @throws GoogleException If the search call fails.
+	 */
+	public function get_product_statuses(): array {
+		$sync_ready_products_ids = $this->product_repository->find_sync_ready_products()->get_product_ids();
+
+		if ( count( $sync_ready_products_ids ) === 0 ) {
+			return $this->calculate_statuses( [] );
+		}
+
+		$offer_ids = array_map(
+			function ( $item ) {
+				return WCProductAdapter::get_offer_id( $item );
+			},
+			$sync_ready_products_ids
+		);
```

Comment:
Seems like the function is unnecessary here, I would have gone for something like this:
```suggestion
		$offer_ids = array_map(
			[ WCProductAdapter::class, 'get_offer_id' ],
			$sync_ready_products_ids
		);
```